pkg/trace/api/otlp_test.go

```diff
@@ -3666,7 +3686,7 @@ func testOTelSpanToDDSpanSetPeerService(enableOperationAndResourceNameV2 bool, t
 				Start:    int64(now),
 				Duration: 200000000,
 				Meta: map[string]string{
-					"deployment.environment":   "prod",
+					"env":                      "prod",
```

Comment:
This is a breaking change, if a customer query their traces on deployment.environment they will no longer be able to see it. This shouldn't be here (at least in this PR)